src/stats.rs

```diff
@@ -0,0 +1,841 @@
+//! Delta Chat has an advanced option
+//! "Send statistics to the developers of Delta Chat".
+//! If this is enabled, a JSON file with some anonymous statistics
+//! will be sent to a bot once a week.
+
+use std::collections::{BTreeMap, BTreeSet};
+
+use anyhow::{Context as _, Result};
+use deltachat_derive::FromSql;
+use pgp::types::PublicKeyTrait;
+use rusqlite::OptionalExtension;
+use serde::Serialize;
+
+use crate::chat::{self, ChatId, ChatVisibility, MuteDuration, ProtectionStatus};
+use crate::config::Config;
+use crate::constants::Chattype;
+use crate::contact::{Contact, ContactId, Origin, import_vcard, mark_contact_id_as_verified};
+use crate::context::{Context, get_version_str};
+use crate::key::load_self_public_keyring;
+use crate::log::LogExt;
+use crate::message::{Message, Viewtype};
+use crate::securejoin::QrInvite;
+use crate::tools::{create_id, time};
+
+pub(crate) const STATISTICS_BOT_EMAIL: &str = "self_reporting@testrun.org";
+const STATISTICS_BOT_VCARD: &str = include_str!("../assets/statistics-bot.vcf");
+const SENDING_INTERVAL_SECONDS: i64 = 3600 * 24 * 7; // 1 week
+const MESSAGE_STATS_UPDATE_INTERVAL_SECONDS: i64 = 4 * 60; // 4 minutes (less than the lowest ephemeral messages timeout)
```

Comment:
There's no minimal value for `DeleteDeviceAfter` in Core, so it makes sense to update message stats right before calling `ephemeral::delete_expired_messages()` (instead of the SMTP loop). `ephemeral_loop()` wakes up once a day by default, this can be slightly decreased to update message stats more frequently if we want